src/main/java/imani/citibike/json/StationLocator.java

```diff
@@ -0,0 +1,45 @@
+package imani.citibike.json;
+
+import imani.citibike.service.CitibikeService;
+import imani.citibike.service.CitibikeServiceFactory;
+import imani.citibike.service.StationUpdaterService;
+
+import java.util.ArrayList;
+
+public class StationLocator {
+    private ArrayList<Station> stationList = new ArrayList<>();
+
+    private CitibikeService citibikeService
+            = new CitibikeServiceFactory().getService();
+    private StationUpdaterService stationUpdaterService;
+
+    public StationLocator(StationUpdaterService stationUpdaterService, CitibikeService citibikeService) {
+        if (stationUpdaterService == null || citibikeService == null) {
+            throw new IllegalArgumentException("Dependencies must not be null");
+        }
+        this.stationUpdaterService = stationUpdaterService;
+        this.citibikeService = citibikeService;
+
+    }
+
+    public Station findClosestStation(double lon, double lat, boolean isBikeSearch, StationUpdaterService sus) {
```

Comment:
You already passed a StationUpdaterService in the constructor, why are you passing in another one?

Reply:
Don't mark my comments as resolved if you aren't actually going to change anything.